controllers/vrg_volrep.go

```diff
@@ -764,10 +764,12 @@ func (v *VRGInstance) undoPVCFinalizersAndPVRetention(pvc *corev1.PersistentVolu
 
 	pvcNamespacedName := types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}
 
-	if pvc.GetAnnotations() != nil && pvc.GetAnnotations()[RestoreAnnotation] == RestoredByRamen {
-		// We created the PVC, delete it
-		if deleted := v.deletePVCIfNotInUse(pvc, log); !deleted {
-			return requeue
+	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary {
```

Comment:
I would just be satisfied with the removal of lines 767,768,769,770.
If the VRG is secondary, this should be unnecessary. We should not have a PVC if the VRG is secondary.

Reply:
@BenamarMk Question I have is, will the PVC get garbage collected when the k8s owner deletion based garbage collection of resources is performed? I thought there was some change to set VRG as the owner for the PVC which may cause this?

Reply:
I am still worried about who owns the PVC that VRG restored once VRG is deleted. Guess I need to run the test and check the app PVC status in between. The worry is due to potential orphan resource being [garbage collected](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#owners-dependents) later by k8s.

Reply:
> I am still worried about who owns the PVC that VRG restored once VRG is deleted. Guess I need to run the test and check the app PVC status in between. The worry is due to potential orphan resource being [garbage collected](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#owners-dependents) later by k8s.

The restored PVC will not be deleted when the VRG is deleted until the Subscription is deleted.